From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Nicolas Iooss <nicolas.iooss_linux@m4x.org>
Date: Thu, 24 Mar 2016 18:34:30 +0100
Subject: [PATCH] {BUG} infiniband: remove use of unitialized variables in
 error

clang reports:
drivers/infiniband/hw/i40iw/i40iw_main.c:605:17: error: variable 'maj_err'
      is uninitialized when used here [-Werror,-Wuninitialized]
                             status, maj_err, min_err);
                                     ^~~~~~~
drivers/infiniband/hw/i40iw/i40iw.h:130:68: note: expanded from macro
      'i40iw_pr_err'
                                                                   ^
include/linux/printk.h:252:33: note: expanded from macro 'pr_err'
        printk(KERN_ERR pr_fmt(fmt), ##__VA_ARGS__)
                                       ^
drivers/infiniband/hw/i40iw/i40iw_main.c:567:13: note: initialize the
      variable 'maj_err' to silence this warning
        u16 maj_err, min_err;
                   ^
                    = 0
drivers/infiniband/hw/i40iw/i40iw_main.c:605:26: error: variable 'min_err'
      is uninitialized when used here [-Werror,-Wuninitialized]
                             status, maj_err, min_err);
                                              ^~~~~~~
drivers/infiniband/hw/i40iw/i40iw.h:130:68: note: expanded from macro
      'i40iw_pr_err'
                                                                   ^
include/linux/printk.h:252:33: note: expanded from macro 'pr_err'
        printk(KERN_ERR pr_fmt(fmt), ##__VA_ARGS__)
                                       ^
drivers/infiniband/hw/i40iw/i40iw_main.c:567:22: note: initialize the
      variable 'min_err' to silence this warning
        u16 maj_err, min_err;
                            ^
                             = 0
---
 drivers/infiniband/hw/i40iw/i40iw_main.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/infiniband/hw/i40iw/i40iw_main.c b/drivers/infiniband/hw/i40iw/i40iw_main.c
index 90e5af21737e..067c719ab7d1 100644
--- a/drivers/infiniband/hw/i40iw/i40iw_main.c
+++ b/drivers/infiniband/hw/i40iw/i40iw_main.c
@@ -601,8 +601,8 @@ static enum i40iw_status_code i40iw_create_cqp(struct i40iw_device *iwdev)
 	cqp_init_info.scratch_array = cqp->scratch_array;
 	status = dev->cqp_ops->cqp_init(dev->cqp, &cqp_init_info);
 	if (status) {
-		i40iw_pr_err("cqp init status %d maj_err %d min_err %d\n",
-			     status, maj_err, min_err);
+		i40iw_pr_err("cqp init status %d\n",
+			     status);
 		goto exit;
 	}
 	status = dev->cqp_ops->cqp_create(dev->cqp, true, &maj_err, &min_err);
-- 
